### ISSUE TITLE
Фантомные раскладки после исправления

### ISSUE THREAD

**Issue by denmkit** [Krot66/LangBarXX#11]
У меня клавиатура Apple c нестандартной для Win раскладкой.
Для полного функционала использую кастомную схему раскладки.
В системе раскладки находятся под номерами:
Рус - a0010419
Eng - a0020409

**Проблема:**
После загрузки системы в панели языков две раскладки:
![Панель языков после перезагрузки](https://github.com/Krot66/LangBarXX/assets/148048243/6c8f0ec5-2e26-4b2a-94ea-6c4c185db847)
После исправления раскладки программой дополнительно появляются две ENG и одна РУС
![Панель языков после исправления раскладки](https://github.com/Krot66/LangBarXX/assets/148048243/3122e6f8-e005-4b00-a72f-cb5ff16af52c)
При этом ни в консоли, ни в языковых параметрах их не появляется:
![Ответ консоли на Get-WinUserLanguageList](https://github.com/Krot66/LangBarXX/assets/148048243/5bdd4bf3-8a2f-40dd-acd2-b83b7415182a)
![Русский](https://github.com/Krot66/LangBarXX/assets/148048243/8e5a0569-06c1-493b-b682-621f735393d2)
![English](https://github.com/Krot66/LangBarXX/assets/148048243/8b2d4d9e-3cb4-4577-be10-cb854dbeefbf)

**Comment by Krot66** [comment]
Первые четыре знака раскладки - это знак дополнительной (а) и ее номер. Дело в том, что для переключения используются только 4 последних разряда и на нормальных раскладках проблем не возникало. А так получается, что вызов раскладки с первыми пустыми разрядами ведет к их удвоению. Исправим

**Comment by denmkit** [comment]
есть еще родные системные раскладки:
0010409 United States-Dvorak
0010419 Russian (Typewriter)
0020409 United States-International
0020419 Russian - Mnemonic

значит c ними тоже может быть эта проблема

**Comment by Krot66** [comment]
Наверное да, хотя их я когда-то ставил и все вроде было нормально. Вообще я надеялся обойти этот вопрос, потому что это темный лес, в который вообще мало кто лезет. Как видно не судьба

**Comment by denmkit** [comment]
В пунто и каком-то еще переключателе видел функцию выбора раскладок для отслеживания и переключения, которые считывались из текущих пользовательских

**Comment by Krot66** [comment]
Считываются они из реестра (HKEY_CURRENT_USER\Keyboard Layout\Preload), вопрос как будет все переключаться

**Comment by denmkit** [comment]
В HKEY_CURRENT_USER\Keyboard Layout\Preload у меня стерильно:
![Снимок экрана 2023-11-12 131508](https://github.com/Krot66/LangBarXX/assets/148048243/6decb97a-9ca1-4fc3-993a-3b8b060e4ca2)

А раскладки почему-то подтягиваются из HKEY_USERS\**.DEFAULT**\Keyboard Layout\Preload:
![Снимок экрана 2023-11-12 111703](https://github.com/Krot66/LangBarXX/assets/148048243/f9cc7ad2-ca4b-4d24-8097-585b02b3cf15)
Но, как видно на скрине, моих раскладок тут нет, к ним прописаны элиасы в HKEY_USERS\.DEFAULT\Keyboard Layout\Substitutes:
![Снимок экрана 2023-11-12 111642](https://github.com/Krot66/LangBarXX/assets/148048243/40c0d03e-323b-48d9-8aa0-fc33af208678)

Пробовал изменять в HKEY_USERS\.DEFAULT, но после перезагрузки всё возвращается...

**Comment by Krot66** [comment]
Хорошо, посмотрим

**Comment by Krot66** [comment]
Посмотрите в списке раскладок в Settings нет ли у вас двух установленных раскладок для английского и русского языка (нужно вызвать опции для каждого языка). Из скриншотов реестра это следует. Просто механизм раскладок Windows достатчно кривой, и в меню они могли появиться не сразу. Скорее всего, вы поставили модифицированные раскладки не удалив стандартные

**Comment by denmkit** [comment]
Нет, там только те раскладки, что на первом скрине реестра. Лишних нет.

**Comment by Krot66** [comment]
Исправлено в версии 1.5.4
